PLAT-121146: Review documetation style for agate components

jeongran1230 opened the pull request:
Enact-DCO-1.0-Signed-off-by: Jeongran Jang (jeongran.jang@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Need to review overall documentation 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Add example tags with example code
- ImageItem, RadioItem, SwitchItem, TabGroup
Update documetation style

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-121146

### Comments

ybsung reviewed (changes requested):
Hello @jeongran1230 
Could check the following list?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/FullscreenPopup/FullscreenPopup.js#L4
Could we add an @example for the FullscreenPopup and the TabbedPanels?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/MediaPlayer/MediaPlayer.js#L721
MediaPlayer's children should have an array. And it would be bette to add a sample source.

AS-IS
```
<source src='' type='' />
 * ```
 * <MediaPlayer>
 *     <source src='' type='' />
 * </MediaPlayer>
 * ```
```

TO-BE
```
 * <MediaPlayer>
 *     <source src={['https://sampleswap.org/mp3/artist/254731/BossPlayer_Your-Right-Here-160.mp3']} type='audio/mp3' />
 * </MediaPlayer>
```

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/RangePicker/RangePicker.js#L5
AS-IS
```
 * <RangePicker defaultValue={70} min={0} max={100}></RangePicker>
```
TO-BE
```
 * <RangePicker defaultValue={70} max={100} min={0} />
```

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Arc/Arc.js#L82
Why is it not 150 but 100?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ArcSlider/ArcSlider.js#L124
Why is it not 30 but 50?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ArcSlider/ArcSliderBehaviorDecorator.js#L81
Why is it not 30 but 50?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ContextualPopupDecorator/ContextualPopup.js#L120
Why is it not 'below center' but 'below'?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ContextualPopupDecorator/ContextualPopup.js#L154
Why is there no `@default false`?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Input/Input.js#L188
Why is the size default value `large`?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/MediaPlayer/MediaPlayer.js#L220
Why does the `repeat` have @default?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Panels.js#L60
Is it correct to write the `closeButtonAriaLabel`'s default value here?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Icon/Icon.js#L39
AS-IS: propTypes: /** @lends agate/Icon.Icon.prototype */ {
TO-BE: propTypes: /** @lends agate/Icon.IconBase.prototype */ {

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/BreadcrumbDecorator.js#L63
Do we need a `@type {Function}` for hoc because other HoCs don't have it?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/TimePicker/TimePickerBase.js#L68
One space before `*`

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/IncrementSlider/IncrementSlider.js#L114 and
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/internal/Picker/Picker.js#L128
Could we switch @default and @type like others?

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/BreadcrumbDecorator.js#L65
`@memberof` would be after `@class`

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Controls.js#L101
Please switch @mixes and @memberof

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/BreadcrumbDecorator.js and
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Viewport.js
@public or @private are missing in some propTypes.

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/TemperatureControl/TemperatureControl.js#L79 and
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/useScroll/Scrollbar.js#L178
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/MediaPlayer/MediaControls.js#L101
We need @default here.

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/SharedStateDecorator.js#L37
`@class` is missing

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/SliderButton/SliderButtonBehaviorDecorator.js#L10
@hoc is missing

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ColorPicker/SwatchButton.js#L27
@extends is missing

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/ApplicationCloseButton.js#L17 and
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Viewport.js#L19
@ui is missing

https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ThemeDecorator/ThemeDecorator.js#L250
no documentaion for ThemeDecorator propTypes.

jeongran1230 commented on ArcPicker/ArcPicker.js:
```diff
@@ -2,7 +2,9 @@
  *
  * @example
- * <ArcPicker backgroundColor="#444444" endAngle={200} foregroundColor="#eeeeee" selectionType="single" startAngle={0} />
+ * <ArcPicker backgroundColor="#cccccc" endAngle={270} startAngle={0}>
```
Default values are not suitable for displaying example component.

jeongran1230 commented on ArcSlider/ArcSlider.js:
```diff
@@ -2,7 +2,7 @@
  *
  * @example
- * <ArcSlider backgroundColor="blue" endAngle={200} foregroundColor="red" radius={150} startAngle={0} step={2} />
+ * <ArcSlider backgroundColor="blue" endAngle={200} radius={150} startAngle={0} step={2} />
```
same reason as above.

jeongran1230 commented on CheckboxItem/CheckboxItem.js:
```diff
@@ -131,6 +131,16 @@ const CheckboxItemDecorator = compose(
 	Skinnable
 );
 
+/**
+ * A Agate-styled item with a checkbox component.
+ *
+ * @class CheckboxItem
+ * @memberof agate/CheckboxItem
+ * @extends agate/CheckboxItem.CheckboxItemBase
+ * @mixes agate/CheckboxItem.CheckboxItemDecorator
+ * @ui
+ * @public
+ */
```
I think either be fine.

jeongran1230 commented on CheckboxItem/CheckboxItem.js:
````diff
@@ -41,7 +41,7 @@ import componentCss from './CheckboxItem.module.less';
  * ```
  *
- * @class CheckboxItem
+ * @class CheckboxItemBase
````
I don't think so.
definition: ` @mixes is used to call out HOCs that may be applied.`

jeongran1230 commented on ImageItem/ImageItem.js:
```diff
@@ -1,6 +1,12 @@
  *
+ * @example
+ * <ImageItem
+ *  src="https://dummyimage.com/64/e048e0/0011ff"
+ *  style={{height: 64, width: 64}}
```
I'm also thinking of indetation. I think one or two space looks better. Any other enact library also use 1 or 2 space.

jeongran1230 commented on Panels/Panels.js:
```diff
@@ -318,6 +317,16 @@ const PanelsDecorator = compose(
 );
 
+/**
+ * An Agate-styled Panels component
```
Why plural? panels is a component.

jeongran1230 commented on RadioItem/RadioItem.js:
```diff
@@ -115,6 +119,16 @@ const RadioItemDecorator = compose(
 );
 
+/**
+ * An Agate-styled Radio Item.
```
will be fixed

jeongran1230 commented on RadioItem/RadioItem.js:
```diff
@@ -24,7 +27,7 @@ import componentCss from './RadioItem.module.less';
+ * @class RadioItemBase
  * @memberof agate/RadioItem
  * @extends agate/Item.Item
  * @mixes agate/RadioItem.RadioItemDecorator
```
will be fixed

jeongran1230 commented on TabGroup/TabGroup.js:
```diff
@@ -195,7 +197,27 @@ const TabGroupDecorator = compose(
+ *   {title: 'Settings', icon: 'setting'},
+ *   {title: 'Theme', icon: 'display'}
+ *  ]}
+ * />
```
will be fixed

jeongran1230 commented on TabGroup/TabGroup.js:
````diff
@@ -195,7 +197,27 @@ const TabGroupDecorator = compose(
+ * Usage:
+ * ```
+ * <TabGroup
+ *  tabPosition={'before'}
+ *  tabs={[
+ *   {title: 'Home', icon: 'home'},
+ *   {title: 'Settings', icon: 'setting'},
+ *   {title: 'Theme', icon: 'display'}
+ *  ]}
````
will be fixed

jeongran1230 commented:
Added 'usage' example code instead of @example.
> Could we add an @example for the FullscreenPopup and the TabbedPanels?

Fixed
>MediaPlayer's children should have an array. And it would be bette to add a sample source.

min -> max is more understandable
> https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/RangePicker/RangePicker.js#L5

Fixed
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Arc/Arc.js#L82
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ArcSlider/ArcSlider.js#L124
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ArcSlider/ArcSliderBehaviorDecorator.js#L81
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ContextualPopupDecorator/ContextualPopup.js#L120
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ContextualPopupDecorator/ContextualPopup.js#L154
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/TimePicker/TimePickerBase.js#L68
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/IncrementSlider/IncrementSlider.js#L114
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/internal/Picker/Picker.js#L128
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/BreadcrumbDecorator.js#L65
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Controls.js#L101
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/useScroll/Scrollbar.js#L178
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/MediaPlayer/MediaControls.js#L101
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/SharedStateDecorator.js#L37
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/SliderButton/SliderButtonBehaviorDecorator.js#L10
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ColorPicker/SwatchButton.js#L27
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/ApplicationCloseButton.js#L17
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Viewport.js#L19
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/ThemeDecorator/ThemeDecorator.js#L250

Not fixed: 
Inherited to internal component's default prop 
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Input/Input.js#L188
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/MediaPlayer/MediaPlayer.js#L220
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Panels.js#L60

Component is private so I don't think to fill in 'private' for each prop
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/BreadcrumbDecorator.js and
https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/Panels/Viewport.js

Bug found
PLAT-125069 https://github.com/enactjs/agate/blob/feature/PLAT-121146-jeongran/TemperatureControl/TemperatureControl.js#L79 

Default values are not suitable for displaying example component.
> I think the simplest example would be the best.

jeongran1230 commented:
move to #398
